src/main/src/main/java/org/geoserver/catalog/CoverageView.java

```diff
@@ -451,4 +507,72 @@ public boolean equals(Object obj) {
         if (selectedResolutionIndex != other.selectedResolutionIndex) return false;
         return true;
     }
+
+    /**
+     * Builds coverage bands for a given catalog store based on the specified composition type.
+     *
+     * @param catalog The catalog containing the coverage store
+     * @param storeId The ID of the coverage store to process
+     * @param compositionType The type of band composition to apply
+     * @param coverageBands List to populate with generated coverage band names
+     * @param inputCoverageBands Map to track input coverage band information
+     * @throws IOException If there is an error reading the coverage store
+     */
+    public static void buildCoverageBands(
+            Catalog catalog,
+            String storeId,
+            CompositionType compositionType,
+            List<String> coverageBands,
```

Comment:
So this is the actual output for the method right? Why is it passed as an argument then?
As far as I can tell it's never read.

Ah, it's because if it it's a Jiffle case, then the input covearge bands are modified too as a side effect. 
Ouch... this is hard to guess looking at the method signature and name. 
Could you have a separate method to collect the input bands from the script?

Reply:
Refactored